src/transport/RendezvousParameters.h

```diff
@@ -42,8 +42,8 @@ class RendezvousParameters
         return *this;
     }
 
-    bool HasDiscriminator() const { return mDiscriminator != 0; }
-    uint16_t GetDiscriminator() const { return mDiscriminator; }
+    bool HasDiscriminator() const { return mDiscriminator != -1; }
+    uint16_t GetDiscriminator() const { return (uint16_t) mDiscriminator; }
```

Comment:
Let's use CPP style cast. Also, can use `CanCastTo` to make sure it'll fit.
Conversely, `HasDiscriminator` can check `CanCastTo` and return false if it can't.

Reply:
Updated it! :)